### ISSUE TITLE
Bug: Suspicious Page not generating as normal

### ISSUE THREAD

**Issue by samuelask** [Tau-ri-Dev/Mod-JSG-1.12.2#353]
The suspicious page is only generating a blank page in every loottable location. Only when manually spawning in structures such as the ZPM obelisk or the Tokra cave does it generate a non generated adress.

**To Reproduce**
1. Easiest way of checking the natural spawning Suspicious pages is to go to the end and look for end cities
2. Find a end city, open up the loot chest
3. You will see that every Suspicious page is empty, just a blank page. However the Uni dialer is spawning as normal.
4. Spawn in ZPM obelisk or another jsg structure that contains a chest by commands, you will see that the Suspicious page spawns in as normal

**Expected behavior**
Your supposed to find a suspicious page and it will automatically change its nbt data to display a non generated adress. Making it so you can explore other dimensions.

**Screenshots**
End city loot
![2024-11-14_17 12 02](https://github.com/user-attachments/assets/3af84799-7e6e-4f1c-9b8c-95466e886f2d)

**Mod version**
JSG 1.12.2
Version 4.11.2.8

**Additional context**
I playing with galacticraft + its extended mod. I have 45 different dimensions that all have non generated gates. It should 100% generate a Suspicious page with a non generated adress. (And it does when manually spawning in structures containing Suspicious pages)

**My whole JSG config (as .zip)**
[jsg.zip](https://github.com/user-attachments/files/17751687/jsg.zip)

**Comment by Fredyman95** [comment]
I think that suspicious page should generates only in JSG structures + in stronghold (this page generate only END gate address)

but maybe we messed something up...

**Comment by samuelask** [comment]
> I think that suspicious page should generates only in JSG structures + in stronghold (this page generate only END gate address)
> 
> but maybe we messed something up...

“end_city_treasure.json” has the “jsg:page_notebook“ in it. So should be intended to spawn a suspicious page?

Maybe “ spawnSusPageInChest” isn’t working properly. It’s supposed to replace the “jsg:page_notebook“ in the chest to a Suspicious page

**Comment by Fredyman95** [comment]
> “end_city_treasure.json” has the “jsg:page_notebook“ in it. So should be intended to spawn a suspicious page?
> 
> Maybe “ spawnSusPageInChest” isn’t working properly. It’s supposed to replace the “jsg:page_notebook“ in the chest to a Suspicious page

page_notebook have more states [data] - empty, filled, suspicious ...

so there is no give command for suspicious as `/give @p suspicious_page 1 0` but if I remember right its `/give @p notebook_page 1 2`

**Comment by samuelask** [comment]
> > “end_city_treasure.json” has the “jsg:page_notebook“ in it. So should be intended to spawn a suspicious page?
> > Maybe “ spawnSusPageInChest” isn’t working properly. It’s supposed to replace the “jsg:page_notebook“ in the chest to a Suspicious page
> 
> page_notebook have more states [data] - empty, filled, suspicious ...
> 
> so there is no give command for suspicious as `/give @p suspicious_page 1 0` but if I remember right its `/give @p notebook_page 1 2`

Yeah but what would be the point in spawning in an empty page as end loot? I think it’s supposed to spawn a suspicious one.

“loot_nether.json” also has the same exact loottable with “jsg:page_notebook“ and that one actually works (changes the notebook page to the intended sus page). It’s the only structure that iv found works consistently as intended. 
I have no idea how all the other structures are broken in this regard, but maybe it’s because the nether gate isn’t randomly generated and is instead spawned when dialing the orlin gate?

**Comment by samuelask** [comment]
Maybe the loottable is named wrong here?

https://github.com/Tau-ri-Dev/Mod-JSG-1.12.2/blob/e37393b60b7c0c222e28e063604ccc134dbf3741/src/main/java/tauri/dev/jsg/worldgen/structures/JSGStructure.java#L285

 case "loot_end": should be “end_city_treasure”

**Comment by Fredyman95** [comment]
> Maybe the loottable is named wrong here?
> 
> https://github.com/Tau-ri-Dev/Mod-JSG-1.12.2/blob/e37393b60b7c0c222e28e063604ccc134dbf3741/src/main/java/tauri/dev/jsg/worldgen/structures/JSGStructure.java#L287
> 
> case "loot_end": should be “end_city_treasure”

If this will be wrong there will will be no loot at all.

I'm still leaning towards it like this: https://github.com/Tau-ri-Dev/Mod-JSG-1.12.2/issues/353#issuecomment-2476866888

**Comment by samuelask** [comment]
Loot would still spawn I think, the end_city_treasure.json would make sure of it. But the “generateLoot” won’t trigger and then the “ spawnSusPageInChest” won’t trigger. 

Also there is no “loot_end“ json file. So that case is literally doing nothing anyway.
